starknet/src/types/proposal.cairo

```diff
@@ -1,16 +1,231 @@
+use sx::types::user_address::UserAddressTrait;
 use clone::Clone;
 use serde::Serde;
 use starknet::ContractAddress;
+use starknet::storage_access::StorePacking;
+use starknet::Store;
+use sx::utils::math::pow;
+use traits::{Into, TryInto};
 use sx::types::{FinalizationStatus, UserAddress};
+use option::OptionTrait;
+use debug::PrintTrait;
 
-#[derive(Clone, Drop, Serde, PartialEq, starknet::Store)]
+const BITMASK_32: u128 = 0xffffffff;
+const BITMASK_64: u128 = 0xffffffffffffffff;
+const BITMASK_128: u128 = 0xffffffffffffffffffffffffffffffff;
+
+const BITMASK_SECOND_U32: u128 = 0xffffffff00000000;
+const BITMASK_THIRD_U32: u128 = 0xffffffff0000000000000000;
+const BITMASK_FOURTH_U32: u128 =
+    0xff000000000000000000000000; // Only 0xff because finalization_status is an u8
+
+const TWO_POWER_32: u128 = 0x100000000;
+const TWO_POWER_64: u128 = 0x10000000000000000;
+const TWO_POWER_96: u128 = 0x1000000000000000000000000;
+
+#[derive(Clone, Drop, Serde, PartialEq)]
 struct Proposal {
     start_timestamp: u32,
     min_end_timestamp: u32,
     max_end_timestamp: u32,
+    finalization_status: FinalizationStatus,
     execution_payload_hash: felt252,
     execution_strategy: ContractAddress,
     author: UserAddress,
-    finalization_status: FinalizationStatus,
     active_voting_strategies: u256
 }
+
+#[derive(Drop, starknet::Store)]
+struct PackedProposal {
+    timestamps_and_finalization_status: u128, // In order: start, min, max, finalization_status
+    execution_payload_hash: felt252,
+    execution_strategy: ContractAddress,
+    author: UserAddress,
+    active_voting_strategies: u256,
+}
+
+impl ProposalStorePacking of StorePacking<Proposal, PackedProposal> {
```

Comment:
really like cairo's approach to packing. super clean and efficient

Reply:
one thing is make sure you `#[derive(starknet::Store)]` and not `#[derive(Store)]`. Found out the hard way...